battles/helpers.py

```diff
@@ -29,11 +28,9 @@ def get_one_a_one_fight_winner(pokemon_1, pokemon_2):
 # battle fight
 def get_battle_winner(battle):
     trainer_team_creator = TrainerTeam.objects.get(
-        Q(battle_related=battle), Q(trainer=battle.trainer_creator)
-    )
+        battle_related=battle, trainer=battle.trainer_creator)
```

Comment:
Having a second look at this, I started thinking that maybe there's a better way to do this. From the `battle` object, you are able to get to both trainers, as you already do here. However, from each trainer, you should be able to get to their teams, instead of querying the DB via the `TrainerTeam` model (e.g. `battle.trainer_creator.teams.filter(battle=battle)`. I'm not sure how much gain (if any) is there in doing this, I'd have to check with someone who has more experience in Django than me, but in any case I thought it was worth mentioning.

For this to happen, however, I'd like to touch on another subject - I think you made a tiny (and very common) confusion with the `related_name` prop of certain model fields. In the `Battle` model you have the `trainer_creator` field, which has 'battle_creator' as its related name. However, this is actually the other way around: the `related_name` property is the name you're going to use, from a `User` instance, to access it's `Battle` instances. So, in this case, the related name should be something like `battles`, so you can access a given user's battles from the `User` instance, like:
``` python
pamella = User.objects.get(email='pamella@vinta.com.br')
pamella.battles.all()  # lists all of Pamella's battles starting from her user
```
This is a very common mistake and even I looked at it a couple times to make sure of what I was saying, so don't worry, haha. The same thing can be done for the `trainer_opponent` field, you can even use the same related name. As for the `trainer_winner` field, I don't think you actually need a related name, so you can just put `related_name='+'`, which means it won't be created.

Also, for all of this to work, in your `TrainerTeam` model, in the `trainer` field, you should add a related name called `teams`, which is how you're gonna access a user's teams from a `User` instance (e.g. `pamella.teams.all()`).

Once you've done all that, these 2 lines could become:
```
trainer_team_creator = battle.trainer_creator.teams.filter(battle=battle).first()
trainer_team_opponent = battle.trainer_opponent.teams.filter(battle=battle).first()
```

Reply:
Yes, I really did make confusion about related names. And I also hadn't thought about doing these queries this way. Thank you again for explaining. I'm gonna use this more often. Done!

PS.: I've tried to set the same `related_name=battles` in the `Battle` model, but I got an error message `Reverse accessor for 'Battle.trainer_creator' clashes with reverse accessor for 'Battle.trainer_opponent'`, so I kept them different.